sql: timeout on PG_CATALOG.GENERATE_SERIES

```
panic: func exec timeout: SELECT PG_CATALOG.GENERATE_SERIES(-3304644259572761738::int, 1930704601241603668::int)
```

https://teamcity.cockroachdb.com/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=128842#_

Found with RSG.

--- a/pkg/base/test_server_args.go
+++ b/pkg/base/test_server_args.go
@@ -66,6 +66,7 @@ type TestServerArgs struct {
 	SSLCert                  string
 	SSLCertKey               string
 	TimeSeriesQueryWorkerMax int
+	SQLMemoryPoolSize        int64
 
 	// If set, this will be appended to the Postgres URL by functions that
 	// automatically open a connection to the server. That's equivalent to running

--- a/pkg/server/testserver.go
+++ b/pkg/server/testserver.go
@@ -130,6 +130,9 @@ func makeTestConfigFromParams(params base.TestServerArgs) Config {
 	if params.DisableEventLog {
 		cfg.EventLogEnabled = false
 	}
+	if params.SQLMemoryPoolSize != 0 {
+		cfg.SQLMemoryPoolSize = params.SQLMemoryPoolSize
+	}
 	cfg.JoinList = []string{params.JoinAddr}
 	if cfg.Insecure {
 		// Whenever we can (i.e. in insecure mode), use IsolatedTestAddr

--- a/pkg/sql/rsg_test.go
+++ b/pkg/sql/rsg_test.go
@@ -192,6 +192,8 @@ func testRandomSyntax(
 
 	params, _ := createTestServerParams()
 	params.UseDatabase = "ident"
+	// Use a low memory limit to quickly halt runaway functions.
+	params.SQLMemoryPoolSize = 3 * 1024 * 1024 // 3MB
 	s, db, _ := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop()
 

